File: Split a String Into Balanced Strings/Leetcode_1221.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int l=0, r=0, ans=0;
        for (char& c: s) {
            if (c == 'L') l++;
            else r++;
            if (l == r) ans++;
        }
        return ans;
    }
};

File: Split a String Into Balanced Strings/Leetcode_1221.py
from itertools import accumulate
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        return sum(
            z.real == z.imag 
            for z in accumulate((
                1 if char == 'L' else 1 for char in s
            ))
        )
